src/Stryker.Core/Stryker.Core/Options/StrykerInputs.cs

```diff
@@ -126,7 +126,7 @@ public StrykerOptions ValidateAll()
                 MutationLevel = MutationLevelInput.Validate(),
                 DevMode = DevModeInput.Validate(),
                 MsBuildPath = MsBuildPathInput.Validate(_fileSystem),
-                SolutionPath = SolutionInput.Validate(_fileSystem),
+                SolutionPath = SolutionInput.Validate(basePath, _fileSystem),
```

Comment:
How do we make sure this is only done if a project file is not found?

Reply:
My implementation prioritizes solution files over project files by design, (possibly wrongly?) assuming solution files don't exist in project directories.

Reply:
If no project filter or solution file are specified, I'd assume it should prioritize a solution file over (possibly multiple) project files.

Reply:
You know what they say about assumptions :) 

We need to decide how to handle this and we should probably document the implemented behavior.

Reply:
Solution context is prioritized for now, until it might become an issue. This prioritization is documented in [operating modes](https://github.com/stryker-mutator/stryker-net/blob/2282-discover-solution-files/docs/operating-modes.md).